--- views.py	2012-12-16 11:27:52.086450380 +0100
+++ views_new.py	2012-12-16 11:37:12.066433837 +0100
@@ -8,7 +8,7 @@
 from django.utils.translation import ugettext_noop
 from django.core.urlresolvers import reverse
 from django.conf import settings
-
+from django.contrib import messages
 from messages.models import Message
 from messages.forms import ComposeForm
 from messages.utils import format_quote
@@ -74,8 +74,7 @@
         form = form_class(request.POST, recipient_filter=recipient_filter)
         if form.is_valid():
             form.save(sender=request.user)
-            request.user.message_set.create(
-                message=_(u"Message successfully sent."))
+	    messages.success(request, _(u"Message successfully sent."))
             if success_url is None:
                 success_url = reverse('messages_inbox')
             if request.GET.has_key('next'):
@@ -108,8 +107,7 @@
         form = form_class(request.POST, recipient_filter=recipient_filter)
         if form.is_valid():
             form.save(sender=request.user, parent_msg=parent)
-            request.user.message_set.create(
-                message=_(u"Message successfully sent."))
+	    messages.success(request, _(u"Message successfully sent."))
             if success_url is None:
                 success_url = reverse('messages_inbox')
             return HttpResponseRedirect(success_url)
@@ -155,7 +153,7 @@
         deleted = True
     if deleted:
         message.save()
-        user.message_set.create(message=_(u"Message successfully deleted."))
+	messages.success(request, _(u"Message successfully deleted."))
         if notification:
             notification.send([user], "messages_deleted", {'message': message,})
         return HttpResponseRedirect(success_url)
@@ -182,7 +180,7 @@
         undeleted = True
     if undeleted:
         message.save()
-        user.message_set.create(message=_(u"Message successfully recovered."))
+	messages.success(request, _(u"Message successfully recovered."))
         if notification:
             notification.send([user], "messages_recovered", {'message': message,})
         return HttpResponseRedirect(success_url)
